ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -936,6 +958,43 @@ generator genBlock m@Model {..} = At <$> frequency
     pointsInDB :: [RealPoint blk]
     pointsInDB = blockRealPoint <$> Map.elems (Model.blocks dbModel)
 
+    -- TODO: sometimes generate fragments that connect to a disconnected block of the ChainDB
+    -- TODO: sometimes generate fragments that exit the ChainDB
```

Comment:
> `-- TODO: sometimes generate fragments that exit the ChainDB`

Do you mean scenarios where the LoE fragment contains blocks that are not yet in the ChainDB? I think that would be very useful to model, as it will be the common case while syncing.

Reply:
Implemented